File: src/main/kotlin/org/smarthata/service/device/heating/HeatingDevice.java
package org.smarthata.service.device.heating;

import java.util.concurrent.atomic.AtomicInteger;
import java.util.concurrent.atomic.AtomicReference;


class HeatingDevice {
    public final String queueActualTemp;
    public final String queueExpectedTemp;
    public final String queueEnabled;
    public final AtomicReference<Double> actualTemp = new AtomicReference<>(0.0);
    public final AtomicReference<Double> expectedTemp;
    public final AtomicInteger enabled = new AtomicInteger(1);

    HeatingDevice(String baseQueue, AtomicReference<Double> expectedTemp) {
        this.queueActualTemp = baseQueue;
        this.queueExpectedTemp = baseQueue + "/in";
        this.queueEnabled = baseQueue + "/enabled";
        this.expectedTemp = expectedTemp;
    }
}

File: src/main/kotlin/org/smarthata/service/device/heating/HeatingDevice.kt
package org.smarthata.service.device.heating

import java.util.concurrent.atomic.AtomicReference

private const val UNDEFINED_TEMP = -127.0

data class HeatingDevice(
    val actualTempQueue: String,
    val expectedTemp: AtomicReference<Double>,
    val nightMode: Boolean = false,
) {
    constructor(actualTempQueue: String, expectedTemp: Double, nightMode: Boolean = false) :
        this(actualTempQueue, AtomicReference(expectedTemp), nightMode)

    val expectedNightTemp = AtomicReference(20.0)

    var actualTemp: Double = UNDEFINED_TEMP
    var enabled: Boolean = true

    val expectedTempQueue = "$actualTempQueue/in"
    val expectedNightTempQueue = "$actualTempQueue/night/in"
    val enabledQueue = "$actualTempQueue/enabled"

}
